README.md

```diff
@@ -5,247 +5,142 @@
-  public static final String OPERATION_DEFINITION = "query DroidDetails {\n"
-      + "  species(id: \"c3BlY2llczoy\") {\n"
-      + "    id\n"
-      + "    name\n"
-      + "    classification\n"
-      + "  }\n"
-      + "}";
-
-  public static final String QUERY_DOCUMENT = OPERATION_DEFINITION;
-
-  private final Operation.Variables variables;
-
-  public DroidDetails() {
-    this.variables = Operation.EMPTY_VARIABLES;
+```
+query FeedQuery($type: FeedType!, $limit: Int!) {
+  feed(type: $type, limit: $limit) {
+    comments {
+      ...FeedCommentFragment
+    }
+    repository {
+      ...RepositoryFragment
+    }
+    postedBy {
+      login
+    }
   }
+}
 
-  @Override
-  public String queryDocument() {
-    return QUERY_DOCUMENT;
+fragment RepositoryFragment on Repository {
+  name
+  full_name
+  owner {
+    login
   }
+}
 
-  @Override
-  public Operation.Variables variables() {
-    return variables;
+fragment FeedCommentFragment on Comment {
+  id
+  postedBy {
+    login
   }
+  content
+}
+```
 
-  public static class Data implements Operation.Data {
-    private static final ResponseFieldMapper<Data> MAPPER = new ResponseFieldMapper<Data>() {
-      private final Field[] FIELDS = {
-        Field.forObject("species", "species", null, true, new Field.ObjectReader<Species>() {
-          @Override public Species read(final ResponseReader reader) throws IOException {
-            return new Species(reader);
-          }
-        })
-      };
-
-      @Override
-      public void map(final ResponseReader reader, final Data instance) throws IOException {
-        reader.read(new ResponseReader.ValueHandler() {
-          @Override
-          public void handle(final int fieldIndex, final Object value) throws IOException {
-            switch (fieldIndex) {
-              case 0: {
-                instance.species = (Species) value;
-                break;
-              }
-            }
-          }
-        }, FIELDS);
-      }
-    };
-    
-    private @Nullable Species species;
+Note: There is nothing Android specific about this query, it can be shared with other GraphQL clients as well
 
-    public Data(ResponseReader reader) throws IOException {
-      MAPPER.map(reader, this);
-    }
+2) You will also need to add a schema to the project. In the sample project you can find the schema `apollo-sample/src/main/graphql/com/example/schema.json`. 
 
-    public Data(@Nullable Species species) {
-      this.species = species;
-    }
+You can find instructions to download your schema using apollo-codegen [HERE](http://dev.apollodata.com/ios/downloading-schema.html)
 
-    public @Nullable Species species() {
-      return this.species;
-    }
+3) Sync Gradle to have Apollo generate the approriate Java classes with nested classes for reading from the network response. In the sample project, a `FeedQuery` Java class is created here `apollo-sample/build/generated/source/apollo/com/example`.
```

Comment:
I think it might make sense to compile your project. The gradle sync also running compile is not guaranteed to always be the case per tools team.